packages/yoastseo/src/parse/build/private/getSentencePositions.js

```diff
@@ -0,0 +1,120 @@
+ * adding mark tags around the sentence. (Same goes for end tags at the start of the sentence, but this should already
+ * never happen, as these tags should be added to the end position of the previous sentence).
+ *
+ * @param {Node[]}				descendantNodes			The descendant nodes.
+ * @param {SourceCodeRange[]}	descendantTagPositions	The positions of the descendant nodes' tags.
+ * @param {number}				sentenceStart			The start position of a sentence.
+ * @param {number}				sentenceEnd				The end position of a sentence.
+ *
+ * @returns {number}	The adjusted end position of the sentence.
+ */
+function adjustSentenceEnd( descendantNodes, descendantTagPositions, sentenceStart, sentenceEnd ) {
+	/*
+	 * If the start position of a descendant's node tag is between the start and end position of the sentence, or is
+	 * the same as the start/end position of the sentence, add the tag's length to the end position of the sentence.
+	 */
+	descendantTagPositions.forEach( ( position ) => {
+		if ( position.startOffset >= sentenceStart && position.startOffset <= sentenceEnd ) {
+			sentenceEnd += ( position.endOffset - position.startOffset );
+		}
+	} );
+	/*
+	 * If the length of a start tag at the end of the sentence was added to the sentenceEnd in the step above,
+	 * remove it. We are using the data from the original descendantNodes array for this check, as the
+	 * descendantTagPositions array doesn't specify whether each tag is an opening or closing tag.
+	 */
+	const startTagAtEndOfSentence = descendantNodes.find( node =>
+		node.sourceCodeLocation.startTag.endOffset === sentenceEnd );
+	if ( startTagAtEndOfSentence ) {
+		const startTagLocation = startTagAtEndOfSentence.sourceCodeLocation.startTag;
+		sentenceEnd = sentenceEnd - ( startTagLocation.endOffset - startTagLocation.startOffset );
+	}
+
+	return sentenceEnd;
+}
+
+/**
+ * Gets the start and end positions of the sentences in the source code.
+ *
+ * @param {Node} 		node  		The paragraph or heading node.
+ * @param {Sentence[]} 	sentences 	The sentences in the node.
+ *
+ * @returns {Sentence[]} The sentences, with their positions in the source code.
+ */
+export default function getSentencePositions( node, sentences ) {
+	/*
+	 * We cannot calculate sentence positions if there are no sentences, or if we don't know the node's source code
+	 * location.
+	 */
+	if (  sentences.length === 0 || ! node.sourceCodeLocation ) {
+		return sentences;
+	}
+	/*
+	 * Set the start position of the first sentence. If the node is an implicit paragraph, which don't have start and
+	 * end tags, set the start position to the start of the node. Otherwise, set the start position to the end of the
+	 * node's start tag.
+	 */
+	let sentenceStart = node.name === "p" && node.isImplicit
+		? node.sourceCodeLocation.startOffset
+		: node.sourceCodeLocation.startTag.endOffset;
+	let sentenceEnd;
+	let descendantNodes = [];
```

Comment:
This line could be removed and then `descendantNodes` could be introduced as a `const`.